com.trollworks.gcs/src/com/trollworks/gcs/utility/Dice.java

```diff
@@ -238,6 +238,15 @@ public void multiply(int multiply) {
         }
     }
 
+    public void percentAdd(double percent) {
+        if (percent != 0) {
+            int average = (mSides + 1) / 2;
+            int base    = (int) Math.floor(((mCount * average) + mModifier) * percent);
+            mCount += base / average;
```

Comment:
You need to guard against division by 0 here. mSides should never come in as 0 or -1, but always safer to not make the assumption.

Reply:
This still doesn't guard against `average` becoming 0 and then being used in division.

Reply:
I knew I forgot something! Thanks.

Reply:
I see now, average never should have been an int, I braindead-ly just wrote it as such.